### PR TITLE
Filter tasks eligilbe for emails by hours_until_closing_date_reminder

### PR DESCRIPTION
**Pull request by alistairclark**

### CODE REVIEW

**Comment by this-andy** [comment]
https://this.tpondemand.com/entity/4829

File: api/endpoints/common/sql_queries.py [hunk 1]
```diff
@@ -378,15 +379,14 @@ def subquery(query):
 
 GET_USERS_ELIGIBLE_FOR_ENDING_SOON_REMINDER_EMAIL = (
     BASE_GET_USERS_ELIGIBLE_FOR_REMINDER_EMAILS
-    + """
-        AND pt.closing_date <= '{datetime_until_task_ends}'
+    + """ 
+        AND (EXTRACT(epoch from (pt.closing_date - NOW()) / 3600)) <= pt.hours_until_closing_date_reminder
```
Comment (this-andy) [inline api/endpoints/common/sql_queries.py:383]: Is this the best way to do this?  It seems a bit obscure.  I was thinking datediff(hour, closing-date, now)

Comment (alistairclark) [reply api/endpoints/common/sql_queries.py:383]: I had a look into this but postgres doesn't have the `DATEDIFF` function. It looks like that's SQL Server. This site suggests a different way of fetching hours from a date, but I'm not sure how much better it is: https://www.sqlines.com/postgresql/how-to/datediff.

I could go with the approach in the above link if you think that's better, but either way I'll add an explanatory comment.

Comment (this-andy) [reply api/endpoints/common/sql_queries.py:383]: Interesting - I didn't realise Postgres didn't implement DATEDIFF.  Rather an odd omission in my view, but we have to deal with it.  I looked at the link and I think I'd favour the approach outlined there - something like
  SELECT DATE_PART('day', '2011-12-30 08:55'::timestamp - '2011-12-30 09:05'::timestamp) * 24 + 
              DATE_PART('hour', '2011-12-30 08:55'::timestamp - '2011-12-30 09:05'::timestamp);

Comment (alistairclark) [reply api/endpoints/common/sql_queries.py]: Have updated that now Andy!

**Review by this-andy — APPROVED** [review]

**Pull request merged**